packages/legend-application-studio/src/stores/editor-state/element-editor-state/connection/PostProcessorEditorState.ts

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {
+  type Mapper,
+  type PostProcessor,
+  type SchemaNameMapper,
+  TableNameMapper,
+} from '@finos/legend-graph';
+import { makeObservable, observable, action } from 'mobx';
+import type { RelationalDatabaseConnectionValueState } from './ConnectionEditorState.js';
+
+export class PostProcessorEditorState {
+  postProcessor: PostProcessor | undefined;
+  connectionValueState: RelationalDatabaseConnectionValueState;
+  selectedMapper: Mapper | undefined;
+  selectedSchema: SchemaNameMapper | undefined;
```

Comment:
why do we have `selectedMapper` here? This is not really the `PostProcessorEditorState`, this is `MapperPostProcessorEditorState`, you should make `PostProcessorEditorState` abstract and have `MapperPostProcessorEditorState` extend it.

Even then, `selectedSchemaMapper` is weird, since from the perspective of `MapperPostProcessorEditorState`, I only care about my `selectedMapper`, I don't know what type of `Mapper` it is.

Reply:
Mhm makes sense, removed the selectedschema and changed to abstract 👍